mlflow/tracking/client.py

```diff
@@ -386,7 +389,7 @@ def get_experiment(self, experiment_id):
         """
         return self._tracking_client.get_experiment(experiment_id)
 
-    def get_experiment_by_name(self, name):
+    def get_experiment_by_name(self, name) -> Experiment:
```

Comment:
I think this API might return `None` if no experiment with the provided name exists, would you be able to double check? If so we may want to fix the docstring as well to indicate that behavior.

Reply:
Thanks for the catch. Yes, it might return None. I'll wrap `Experiment` with `Optional`.